### ISSUE TITLE
[visibility bug] HTML-Tag "<i class='icon-arrow-up'></i>"does not work

### ISSUE THREAD

**Issue by htcfreek** [XavierBerger/RPi-Monitor#309]
Hello.

With the html tag `<i class='icon-arrow-up'></i>` you can show symbols in the text on the html page. But this tag won't work for me in rpimonitor 2.12-r0. I use the original text form the template `/etc/rpimonitor/template/network.conf`.
I have seen this on Microsoft Edge (Chromium), Firefox and Chromium.

Is this a browser bug or an rpimonitor bug?

Thanks for your help.
htcfreek

**Comment by wrapper** [comment]
+1 I also don't see it

**Comment by GlynHu** [comment]
Me three!

This affects both network.conf & wlan.conf

The stats and layout all look good although I have wi-fi disabled for the time being but no up/down arrows are displayed.

![Untitled](https://user-images.githubusercontent.com/78040656/105914900-e9a72080-6026-11eb-876f-291db3668787.jpg)

Most screenshots I see have the arrows missing although I have seen a few where the blue up/down arrows can be seen adjacent to the numbers reported.

The respective calls appear identical in both .conf files and looks like this rerspectively;

web.status.1.content.8.line.1="Ethernet Sent: <b>"+KMG(data.net_send)+"<i class='icon-arrow-up'></i></b> Received: <b>"+KMG(Math.abs(data.net_received)) + "<i class='icon-arrow-down'></i></b>"

web.status.1.content.9.line.1="WiFi Sent: <b>"+KMG(data.wifi_send)+"<i class='icon-arrow-up'></i></b> Received: <b>"+KMG(Math.abs(data.wifi_received)) + "<i class='icon-arrow-down'></i></b>"

Not sure where it might be looking for the arrows but they don't exist in the /web/img folder.

Obviously not mission critical but it is in there for a reason so should work and would look pretty if it did!

Again if there is anything I can do to help then please let me know.

Kind regards,
-=Glyn=-

**Comment by Drarig29** [comment]
This is linked to this: https://stackoverflow.com/questions/25466519/bootstrap-3-2-0-glyphicons-not-working

This is the version of Bootstrap used by RPi-Monitor.

So you can replace `icon-arrow-up` with `glyphicon glyphicon-arrow-up`.

Here is the line I use in `network.conf`:

```html
web.status.1.content.8.line.1="<p>Ethernet Sent: <b>"+KMG(data.net_send)+"<i class='glyphicon glyphicon-arrow-up'></i></b></p><p>Received: <b>"+KMG(Math.abs(data.net_received)) + "<i class='glyphicon glyphicon-arrow-down'></i></b></p>"
```

**Comment by sgrayban** [comment]
@Drarig29 awesome fix!

![image](https://user-images.githubusercontent.com/1351910/147304227-2e7f6cf0-19c6-4c1b-b8d5-e4e1891f72fe.png)

**Comment by GlynHu** [comment]
Yup...almost a year on from my original post I stumbled across the fix...many thanks @Drarig29 

Where are these glyphicons stored as I mentioned above there seemed to be some rather nice, larger blue icons I have seen on others screenshots?

Thanks & kind regards,
-=Glyn=-

**Comment by GlynHu** [comment]
> @Drarig29 awesome fix!
> 
> ![image](https://user-images.githubusercontent.com/1351910/147304227-2e7f6cf0-19c6-4c1b-b8d5-e4e1891f72fe.png)

I like the Net Speed icon & layout @sgrayban - Can I ask how you did this please?

It's been a looong time since I installed, configured & played with RPIMonitor and I've forgotten more than I ever learned! :)

Thanks & kind regards,
-=Glyn=-

**Comment by Drarig29** [comment]
Here is the list of glyphicons: https://getbootstrap.com/docs/3.3/components/

But the bigger icons like "Network" and "Net speed" do not come from this.

You'll need to find them by yourself and store them in `/usr/share/rpimonitor/web/img` I believe (according to [this folder](https://github.com/XavierBerger/RPi-Monitor/tree/develop/src/usr/share/rpimonitor/web/img)).

After this, you can refer to this images by their filename.

Example with [wifi.png](https://github.com/XavierBerger/RPi-Monitor/blob/develop/src/usr/share/rpimonitor/web/img/wifi.png):

https://github.com/XavierBerger/RPi-Monitor/blob/53a95542e13f0289e117da3d6a18559604b25c0f/src/etc/rpimonitor/template/wlan.conf#L23

---

Take this with a grain of salt, I'm just guessing. I don't have my raspberry pi anymore.

**Comment by sgrayban** [comment]
> > @Drarig29 awesome fix!
> > ![image](https://user-images.githubusercontent.com/1351910/147304227-2e7f6cf0-19c6-4c1b-b8d5-e4e1891f72fe.png)
> 
> I like the Net Speed icon & layout @sgrayban - Can I ask how you did this please?
> 
> It's been a looong time since I installed, configured & played with RPIMonitor and I've forgotten more than I ever learned! :)
> 
> Thanks & kind regards, -=Glyn=-

Here is my icon
![image](https://user-images.githubusercontent.com/1351910/149891779-2b43d164-884a-4120-b075-ec4c64fc3e37.png)

**Comment by GlynHu** [comment]
Thanks both for taking the time to reply to my post.

I would also like to know the lines needed and how to call & display the Net Speed Ping, Download & Upload info if possible please? 

Thanks & kind regards,
-=Glyn=-

**Comment by sgrayban** [comment]
See this post https://github.com/XavierBerger/RPi-Monitor/issues/383

**Comment by GlynHu** [comment]
Great…thanks Scott I’ll give it a go.

Thanks & kind regards,

-=Glyn=-
